tonic/models/vic/grid_params.py

```diff
@@ -661,64 +1218,172 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
                 for y, x in pyzip(ymask, xmask):
                     out_dict[var][:, :, y, x] = fill_val
 
+            elif mydict[var].ndim == 4:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                out_dict[var] = np.ma.zeros((j, k, l, ysize, xsize),
+                                            dtype=dtype)
```

Comment:
Can't you just return a tuple and put this into the `np.ma.zeros` line, e.g. `out_dict[var] = np.ma.zeros((mydict[var].shape), ysize, xsize`? I guess you might need to convert the tuple to an array but I think it'd be cleaner to combine these two steps...